app/views/admin/AdministerUserModal.coffee

```diff
@@ -2,47 +2,37 @@ ModalView = require 'views/core/ModalView'
 template = require 'templates/admin/administer-user-modal'
 User = require 'models/User'
 Prepaid = require 'models/Prepaid'
+StripeCoupons = require 'collections/StripeCoupons'
+forms = require 'core/forms'
+Prepaids = require 'collections/Prepaids'
 
 module.exports = class AdministerUserModal extends ModalView
-  id: "administer-user-modal"
+  id: 'administer-user-modal'
   template: template
 
   events:
-    'click #save-changes': 'onSaveChanges'
+    'click #save-changes': 'onClickSaveChanges'
     'click #add-seats-btn': 'onClickAddSeatsButton'
 
-  constructor: (options, @userHandle) ->
-    super(options)
-    @user = @supermodel.loadModel(new User({_id:@userHandle}), {cache: false}).model
-    options = {cache: false, url: '/stripe/coupons'}
-    options.success = (@coupons) =>
-    @couponsResource = @supermodel.addRequestResource('coupon', options)
-    @couponsResource.load()
-
-  getRenderData: ->
-    c = super()
+  initialize: (options, @userHandle) ->
+    @user = new User({_id:@userHandle})
+    @supermodel.trackRequest @user.fetch({cache: false})
+    @coupons = new StripeCoupons()
+    @supermodel.trackRequest @coupons.fetch({cache: false})
+    @prepaids = new Prepaids()
+    @supermodel.trackRequest @prepaids.fetchByCreator(@userHandle)
+    
+  onLoaded: ->
+    # TODO: Figure out a better way to expose this info, perhaps User methods?
     stripe = @user.get('stripe') or {}
-    c.free = stripe.free is true
-    c.freeUntil = _.isString(stripe.free)
-    c.freeUntilDate = if c.freeUntil then stripe.free else new Date().toISOString()[...10]
-    c.coupon = stripe.couponID
-    c.coupons = @coupons or []
-    for coupon in c.coupons
-      bits = [coupon.id]
-      if coupon.percent_off
-        bits.push "(#{coupon.percent_off}% off)"
-      else if coupon.amount_off
-        bits.push "($#{coupon.amount_off} off)"
-      if coupon.duration
-        bits.push "(duration: #{coupon.duration})"
-      if coupon.redeem_by
-        bits.push "(redeem by: #{moment(coupon.redeem_by).format('lll')}"
-      coupon.format = bits.join(' ')
-    c.none = not (c.free or c.freeUntil or c.coupon)
-    c.user = @user
-    c
-
-  onSaveChanges: ->
+    @free = stripe.free is true
+    @freeUntil = _.isString(stripe.free)
```

Comment:
Yeah, me too. They're holdovers from how the modal was implemented before; it could use more thorough, thoughtful refactoring.